x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/search/action/AsyncStatusResponse.java

```diff
@@ -0,0 +1,200 @@
+import org.elasticsearch.common.xcontent.StatusToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.rest.action.RestActions;
+import org.elasticsearch.xpack.core.async.AsyncResponse;
+
+import java.io.IOException;
+
+import static org.elasticsearch.rest.RestStatus.OK;
+
+/**
+ * A response of an async search request.
+ */
+public class AsyncStatusResponse extends ActionResponse implements StatusToXContentObject, AsyncResponse<AsyncStatusResponse> {
+    private final String id;
+    private final boolean isRunning;
+    private final long startTimeMillis;
+    private final long expirationTimeMillis;
+    private final int totalShards;
+    private final int successfulShards;
+    private final int skippedShards;
+    private final int failedShards;
+
+    private AsyncStatusResponse(String id,
+            boolean isRunning,
+            long startTimeMillis,
+            long expirationTimeMillis,
+            int totalShards,
+            int successfulShards,
+            int skippedShards,
+            int failedShards) {
+        this.id = id;
+        this.isRunning = isRunning;
+        this.startTimeMillis = startTimeMillis;
+        this.expirationTimeMillis = expirationTimeMillis;
+        this.totalShards = totalShards;
+        this.successfulShards = successfulShards;
+        this.skippedShards = skippedShards;
+        this.failedShards = failedShards;
+    }
+
+    /**
+     * Creates a new {@link AsyncStatusResponse} for a running search.
+     *
+     * @param id The id of the async search.
+     * @param startTimeMillis The start date of the search in milliseconds since epoch.
+     * @param expirationTimeMillis The expiration date of the search in milliseconds since epoch.
+     * @param totalShards The total number of shards the request search is executed on
+     * @param successfulShards The number of shards the request is completed on
+     * @param skippedShards The number of skipped shards
+     * @param failedShards The number of shards that failed to executed the request
+     */
+    public AsyncStatusResponse(String id,
+            long startTimeMillis,
+            long expirationTimeMillis,
+            int totalShards,
+            int successfulShards,
+            int skippedShards,
+            int failedShards) {
+        this(id, true, startTimeMillis, expirationTimeMillis, totalShards, successfulShards, skippedShards, failedShards);
+    }
+
+    /**
+     * Creates a new {@link AsyncStatusResponse} for a completed search.
+     *
+     * @param id The id of the async search.
+     * @param expirationTimeMillis – expiration time in milliseconds
+     * @return status of the completed async search
+     */
+    public static AsyncStatusResponse getCompletedSearchStatusResponse(String id, long expirationTimeMillis) {
+        return new AsyncStatusResponse(id, false, 0, expirationTimeMillis, 0, 0, 0, 0);
```

Comment:
Why would we set the stats to 0 on a completed search ? That seems wrong to me.

Reply:
addressed in 0775cf1